tests/Microsoft.VisualStudio.ProjectSystem.Managed.UnitTests/Mocks/ILanguageServiceTelemetryServiceFactory.cs

```diff
@@ -0,0 +1,16 @@
+﻿// Licensed to the .NET Foundation under one or more agreements. The .NET Foundation licenses this file to you under the MIT license. See the LICENSE.md file in the project root for more information.
+
+using Moq;
+
+namespace Microsoft.VisualStudio.Telemetry
+{
+    internal static class ILanguageServiceTelemetryServiceFactory
+    {
+        public static ILanguageServiceTelemetryService Create()
+        {
+            var mock = new Mock<ILanguageServiceTelemetryService>();
+
+            return mock.Object;
```

Comment:
Is there a reason we prefer this boilerplate pattern over simply calling `new Mock` at the callsite?

Reply:
I did this to be consistent with the other tests I looked at. My interpretation was that the goal of this usage of the factory pattern is to abstract the exact manner in which an implementation of the returned interface is created so that, for example, a switch from XUnit to Moq to something else doesn't need the individual tests to be modified. Perhaps other folks might have some thoughts about this?

---
In reply to: [567169220](https://github.com/dotnet/project-system/pull/6880#discussion_r567169220) [](ancestors = 567169220)